stdlib/public/core/Integers.swift.gyb

```diff
@@ -3296,10 +3296,15 @@ public struct ${Self}
   /// - Parameter source: A floating-point value to convert to an integer.
   @_transparent
   public init?(exactly source: ${FloatType}) {
-    self._value = Builtin.fpto${u}i_FPIEEE${FloatBits}_${BuiltinName}(source._value)
-    if ${FloatType}(self) != source {
+    if _slowPath(source <= ${str(lower)}.0 || source >= ${str(upper)}.0) {
```

Comment:
These bounds are _exclusive_ so the bounds themselves are not valid integers. For example, the lower bound of an unsigned integer is -1. The code in the preceding method is performing the opposite check so the comparisons are inverted (i.e. > instead of <= and < instead of >=).